stm32-modules/include/thermocycler-refresh/thermocycler-refresh/tmc2130_registers.hpp

```diff
@@ -0,0 +1,380 @@
+/**
+ * @file tmc2130_registers.hpp
+ * @brief Contains register mapping information for the TMC2130 motor
+ * driver IC.
+ *
+ */
+#pragma once
+
+#include <concepts>
+#include <cstdint>
+
+namespace tmc2130 {
+
+// Register mapping
+
+enum class Registers {
+    GCONF = 0x00,
+    GSTAT = 0x01,
+    IOIN = 0x04,
+    IHOLD_IRUN = 0x10,
+    TPOWERDOWN = 0x11,
+    TSTEP = 0x12,
+    TPWMTHRS = 0x13,
+    TCOOLTHRS = 0x14,
+    THIGH = 0x15,
+    XDIRECT = 0x2D,
+    VDCMIN = 0x33,
+    CHOPCONF = 0x6C,
+    COOLCONF = 0x6D,
+    DCCTRL = 0x6E,
+    DRVSTATUS = 0x6F,
+    PWMCONF = 0x70,
+    ENCM_CTRL = 0x72,
+    MSLUT_0 = 0x60,
+    MSLUT_1 = 0x61,
+    MSLUT_2 = 0x62,
+    MSLUT_3 = 0x63,
+    MSLUT_4 = 0x64,
+    MSLUT_5 = 0x65,
+    MSLUT_6 = 0x66,
+    MSLUT_7 = 0x67,
+    MSLUTSEL = 0x68,
+    MSLUTSTART = 0x69,
+    MSCNT = 0x6A,
+    MSCURACT = 0x6B,
+    PWM_SCALE = 0x71,
+    LOST_STEPS = 0x73,
+};
+
+/** Template concept to constrain what structures encapsulate registers.*/
+template <typename Reg>
+concept TMC2130Register = requires(Reg& r, uint64_t value) {
+    // Struct has a valid register address
+    std::same_as<decltype(Reg::address), Registers&>;
+    // Struct has a bool saying whether it can be read
+    std::same_as<decltype(Reg::readable), bool>;
+    // Struct has a bool saying whether it can be written
+    std::same_as<decltype(Reg::writable), bool>;
+    // Struct has an integer with the total number of bits in a register.
+    // This is used to mask the 64-bit value before writing to the IC.
+    std::integral<decltype(Reg::bitlen)>;
```

Comment:
given what `bitlen` is used for, does it make more sense for it to be `value_mask` or something and either written as a mask directly or `consteval`ed from a bitshift?